core/shared/src/main/scala/scalaz/zio/IO.scala

```diff
@@ -682,7 +682,12 @@ sealed abstract class IO[+E, +A] extends Serializable { self =>
    * Runs this action in a new fiber, resuming when the fiber terminates.
    */
   final def run: IO[Nothing, ExitResult[E, A]] =
-    redeem0(cause => IO.now(ExitResult.failed(cause)), succ => IO.now(ExitResult.succeeded(succ)))
+    new IO.Redeem[E, Nothing, A, ExitResult[E, A]](
+      self,
+      cause => IO.now(ExitResult.failed(cause)),
+      succ => IO.now(ExitResult.succeeded(succ)),
+      recoverFromInterruption = true
+    )
```

Comment:
That will be wicked fast. 😆 

Can you add a test to show this works? i.e. that if you do `io.run.flatMap(_ => promise.complete(42))` the promise is actually completed?